src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +16,32 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            actor.setId((Long) session.getIdentifier(actor));
+            transaction.commit();
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't save actor to DB. Actor: " + actor, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
```

Comment:
Let's use try with resources, because Session implements Autocloseable interface
```suggestion
        Transaction transaction = null;
        try(Session session = factory.openSession()) {
            session = factory.openSession();
            transaction = session.beginTransaction();
            session.persist(actor);
            actor.setId((Long) session.getIdentifier(actor));
            transaction.commit();
        } catch (HibernateException e) {
            if (transaction != null) {
                transaction.rollback();
            }
            throw new DataProcessingException("Can't save actor to DB. Actor: " + actor, e);
        }
```

Reply:
It's not recommended to use try with resources with session
https://mate.academy/learn/java-hibernate/java-entities-and-transactions?learnItemsFilter=All&section=video&videoId=1043